node/src/test/scala/com/wavesplatform/state/diffs/smart/predef/ScriptVersionsTest.scala

```diff
@@ -92,11 +104,56 @@ class ScriptVersionsTest extends FreeSpec with PropertyChecks with Matchers with
 
   "ScriptV3" - {
     "disallow transactionById" in {
-        eval[EVALUATED](txById, V3) should produce("Can't find a function 'transactionById'")
+      eval[EVALUATED](txById, V3) should produce("Can't find a function 'transactionById'")
     }
 
     "add transferTransactionById" in {
       eval[EVALUATED](transferTxById, V3) shouldBe Testing.evaluated(true)
     }
   }
+
+  "ScriptV4" - {
+    "DataTransaction entry mapping" in {
+      def compile(scriptText: String) =
+        ScriptCompiler.compile(scriptText, ScriptEstimatorV2).explicitGet()._1
+
+      def script(dApp: Boolean) =
+        compile(
+          s"""
+             | {-# STDLIB_VERSION 4       #-}
+             | {-# SCRIPT_TYPE    ACCOUNT #-}
+             | {-# CONTENT_TYPE ${if (dApp) "DAPP" else "EXPRESSION"}  #-}
+             |
+             | ${if (dApp) "@Verifier(tx) \n func verify() = " else ""}
+             | match tx {
+             |   case d:DataTransaction =>
+             |       d.data.size() == 1 &&
+             |       match d.data[0] {
+             |         case entry: StringEntry =>
+             |           entry.key == "key" &&
+             |           entry.value == "value"
+             |         case _: IntegerEntry =>
+             |           throw("unexpected IntegerEntry")
+             |         case _: BinaryEntry =>
+             |           throw("unexpected BinaryEntry")
+             |         case _: BooleanEntry =>
+             |           throw("unexpected BooleanEntry")
+             |       }
+             |   case _ =>
+             |     sigVerify(tx.bodyBytes, tx.proofs[0], tx.senderPublicKey)
+             | }
+       """.stripMargin
+        )
+
+      val tx = dataTransactionGen.sample.get.copy(data = Seq(StringDataEntry("key", "value")))
```

Comment:
please add cases for Integer/Binary/Boolean/DeleteEntry mapping

Reply:
Added excluding DeleteEntry, it is not supported until https://github.com/wavesplatform/Waves/pull/3266 will be merged